packages/core/test/typegenTypes.test.ts

```diff
@@ -967,4 +967,58 @@ describe('typegen types', () => {
       }
     );
   });
+
+  it("should not provide a loose type for `onReceive`'s argument as a default", () => {
+    interface TypesMeta extends TypegenMeta {
+      eventsCausingServices: {
+        fooService: 'FOO';
+      };
+    }
+
+    createMachine(
+      {
+        tsTypes: {} as TypesMeta,
+        schema: {
+          events: {} as { type: 'FOO' } | { type: 'BAR' }
+        }
+      },
+      {
+        services: {
+          fooService: (ctx, ev) => (send, onReceive) => {
+            onReceive((event) => {
+              ((_accept: string) => {})(event.type);
+              // @ts-expect-error
+              event.unknown;
+            });
+          }
+        }
+      }
+    );
+  });
+
+  it("should allow specifying `onReceive`'s argument type manually", () => {
```

Comment:
This test isn't really showcasing a completely type-safe behavior. This relies on the introduced `bivarianceHack` - while it's unsafe it feels like a balance between correct & friendly to the user.

In the future, I hope to figure out how to make communication between actors completely type-safe.